hazelcast/src/main/java/com/hazelcast/map/impl/MapConfigValidator.java

```diff
@@ -36,36 +38,54 @@ private MapConfigValidator() {
-        if (NATIVE == inMemoryFormat) {
-            throw new IllegalArgumentException("NATIVE storage format is supported in Hazelcast Enterprise only. "
-                    + "Make sure you have Hazelcast Enterprise JARs on your classpath!");
-        }
+    public static void checkMapConfig(MapConfig mapConfig) {
+        checkNotNative(mapConfig.getInMemoryFormat());
+
+        logIgnoredConfig(mapConfig);
     }
 
     /**
-     * Checks preconditions to create a map proxy.
+     * Checks preconditions to create a map proxy with Near Cache.
      *
-     * @param mapConfig the mapConfig
+     * @param nearCacheConfig the {@link NearCacheConfig}
+     * @param isClient        {@code true} if the config is for a Hazelcast client, {@code false} otherwise
      */
-    public static void checkMapConfig(MapConfig mapConfig) {
-        checkNotNative(mapConfig.getInMemoryFormat());
+    public static void checkNearCacheConfig(NearCacheConfig nearCacheConfig, boolean isClient) {
+        checkNotNative(nearCacheConfig.getInMemoryFormat());
 
-        logIgnoredConfig(mapConfig);
+        logIgnoredConfig(nearCacheConfig, isClient);
+    }
+
+    /**
+     * Throws {@link IllegalArgumentException} if the supplied {@link InMemoryFormat} is {@link InMemoryFormat#NATIVE}
+     *
+     * @param inMemoryFormat supplied inMemoryFormat
+     */
+    private static void checkNotNative(InMemoryFormat inMemoryFormat) {
+        if (inMemoryFormat == NATIVE && !getBuildInfo().isEnterprise()) {
+            throw new IllegalArgumentException("NATIVE storage format is supported in Hazelcast Enterprise only."
+                    + " Make sure you have Hazelcast Enterprise JARs on your classpath!");
+        }
     }
 
     @SuppressWarnings("deprecation")
     private static void logIgnoredConfig(MapConfig mapConfig) {
-        if (DEFAULT_MIN_EVICTION_CHECK_MILLIS != mapConfig.getMinEvictionCheckMillis()
-                || DEFAULT_EVICTION_PERCENTAGE != mapConfig.getEvictionPercentage()) {
+        if (mapConfig.getMinEvictionCheckMillis() != DEFAULT_MIN_EVICTION_CHECK_MILLIS
+                || mapConfig.getEvictionPercentage() != DEFAULT_EVICTION_PERCENTAGE) {
+            LOGGER.warning("As of version 3.7 `minEvictionCheckMillis` and `evictionPercentage`"
+                    + " are deprecated due to a change of the eviction mechanism."
+                    + " The new eviction mechanism uses a probabilistic algorithm based on sampling."
+                    + " Please see documentation for further details.");
+        }
+    }
 
-            LOGGER.warning("As of version 3.7, `minEvictionCheckMillis` and `evictionPercentage` "
-                    + "are deprecated due to the eviction mechanism change. New eviction mechanism "
-                    + "uses a probabilistic algorithm based on sampling. Please see documentation for further details");
+    private static void logIgnoredConfig(NearCacheConfig nearCacheConfig, boolean isClient) {
```

Comment:
We could make it a general `ConfigValidator`, since there is some map specific checks in here, some `NearCacheConfig` checks and maybe we'll add other ones in the future.